hr_holidays_hour/report/hr_holidays_report.py

```diff
@@ -38,15 +40,32 @@ def init(self):
         cr.execute("SELECT pg_get_viewdef(%s, true)", (self._table,))
         view_def = cr.fetchone()[0]
         view_def = view_def.replace("number_of_days", "number_of_hours")
-        view_def = view_def.replace(
-            "hhs.name as leave_type",
-            "{} hhs.name as leave_type".format(self._holidays_hour_select()),
-        )
+        holidays_hour_select = self._holidays_hour_select()
+        if holidays_hour_select not in view_def:
+            # The 'leave_type' query could have case-sensitive expressions
+            # It's better to use re here than the crude replace
+            # (we basically search for "hhs.name AS leave_type")
+            leave_type_query_part = re.compile(
+                '.*,(.*?leave_type).*'
+            ).match(' '.join(view_def.split('\n'))).groups()[0]
+            view_def = view_def.replace(
+                leave_type_query_part.strip(),
+                "{},\n{}".format(
+                    self._holidays_hour_select(),
+                    leave_type_query_part
+                ),
+            )
         if view_def[-1] == ';':
             view_def = view_def[:-1]
-        view_def += self._holidays_hour_group_by()
+        holidays_hour_group_by = self._holidays_hour_group_by()
+        if holidays_hour_group_by not in view_def:
+            view_def += self._holidays_hour_group_by()
         # Re-create view
         tools.drop_view_if_exists(cr, self._table)
-        cr.execute("create or replace view {} as ({})".format(
-            self._table, view_def,
+        sql = SQL('CREATE OR REPLACE VIEW {} as {}')
```

Comment:
Using `AsIs` method you can do it directly. See for example: https://github.com/OCA/OpenUpgrade/pull/1589/files#diff-438d8381a8d27b194556dd3c7cbbe766R140